lexxpluss_apps/src/pgv_controller.cpp

```diff
@@ -145,9 +137,9 @@ class pgv_controller_impl {
         send(req, sizeof req);
         wait_data(23);
         uint8_t buf[64];
-        if (int n{recv(buf, sizeof buf)}; n < 23 || !validate(buf + 2, 21))
+        if (int n{recv(buf, sizeof buf)}; n < 23 || !validate(buf + 2, 21)) // 21byte + echo back 2byte (may be)
             return false;
-        decode(buf + 2, data);
+        memcpy(data.rawdata, buf, 21);
```

Comment:
```suggestion
        memcpy(data.rawdata, buf + 2, 21);
```
The first 2 bytes of `buf` may contain the echo back.
What you are validating above is `buf + 2`.
`buf`の先頭2バイトはecho backが格納されているのではないでしょうか。
上でvalidateしているのは`buf + 2`です。